Microsoft.Toolkit.Sample.Wpf.App/Program.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Windows.UI.Xaml.Markup;
+
+namespace Microsoft.Toolkit.Sample.Wpf.App
+{
+    public class MyControl : Toolkit.Wpf.UI.XamlHost.WindowsXamlHostBase
+    {
+        static MyControl()
+        {
+            if (MetadataContainer != null)
+            {
+                MetadataContainer.MetadataProviders.Add(new MyMetadataProvider());
+            }
+            //else
+            //{
+            //    throw new InvalidOperationException($"{typeof(MyControl).Name} only supported when for Windows.UI.Xaml.Application instances that implements {typeof(Win32.UI.XamlHost.IXamlMetadataContainer).FullName}");
+            //}
+        }
+
+        public MyControl()
+        {
+        }
+    }
+
+    public sealed class MyMetadataProvider : IXamlMetadataProvider
+    {
+        public IXamlType GetXamlType(Type type)
+        {
+            return null;
+        }
+
+        public IXamlType GetXamlType(string fullName)
+        {
+            return null;
+        }
+
+        public XmlnsDefinition[] GetXmlnsDefinitions()
+        {
+            return System.Linq.Enumerable.Empty<XmlnsDefinition>().ToArray();
+        }
+    }
+
+    public static class Program
+    {
+        enum StartupKind
```

Comment:
Do we have docs on this different startup scenarios?

Reply:
I am actively working with Miguel to validate the new API and provide documentation.

---
In reply to: [259952285](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/85#discussion_r259952285) [](ancestors = 259952285)